connector/src/main/java/org/jboss/as/connector/subsystems/datasources/Constants.java

```diff
@@ -227,18 +227,15 @@
     static final SensitiveTargetAccessConstraintDefinition DS_SECURITY_DEF = new SensitiveTargetAccessConstraintDefinition(DS_SECURITY);
 
     static final SimpleAttributeDefinition DEPLOYMENT_NAME = SimpleAttributeDefinitionBuilder.create("deployment-name", ModelType.STRING)
-            .setAllowExpression(true)
```

Comment:
This and JDBC_COMPLIANT are used in two different ways -- as an attribute in the jdbc-driver resource, and then as part of the reply to get-installed-driver/installed-driver-list.

It seems like different AttributeDefinitions should be used for these two ways.  This one can be used for get-installed-driver/installed-driver-list. Expressions make no sense there. The other AD can be used for the jdbc-driver resource. It should allow expressions. But AFAICT those attributes are cruft, not actually used in any service. So add setDeprecationData to the AD builder. And then in another WF release or two,  remove them.

Reply:
done